### PR TITLE
fix(): slicegw health status for single cluster

### PR DESCRIPTION
**Pull request by Rahul-D78**
For a single cluster, sliceGW pods are missing.
We need to handle component health status for two cases 1. single cluster 2. multiple cluster

case 1: we check if sliceGW deployment is present, expected no slicegw deployments hence, mark healthystate

case 2: we check if sliceGW deployment is present, expected 1 slicegw deployment, hence, list all pods and check if all are in Running state else mark as unhealthyState

### CODE REVIEW

**Review by gourishkb — APPROVED** [review]

File: pkg/hub/controllers/cluster/reconciler.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func (r *Reconciler) updateNodeIps(ctx context.Context, cr *hubv1alpha1.Cluster)
 		// the current nodeIpList (nodeIpList contains list of externalIPs or internalIPs)
 		// if the nodeIP is no longer available, we update the cluster CR on controller cluster
 		if cr.Status.NodeIPs == nil || len(cr.Status.NodeIPs) == 0 ||
-			!validatenodeips(nodeIPs, cr.Status.NodeIPs) {
+			!validatenodeips(nodeIPs, cr.Status.NodeIPs) || !isValidNodeIpList(nodeIPs) {
 			log.Info("Mismatch in node IP", "IP in use", cr.Status.NodeIPs, "IP to be used", nodeIPs)
```
Comment (YachikaRalhan) [inline pkg/hub/controllers/cluster/reconciler.go:280]: why is this needed? validatenodeips function is already returning false if the elements of array are not same.
